src/polyrith.lean

```diff
@@ -74,7 +89,7 @@ meta def has_val: ℕ → (expr × ℕ) → tactic unit
 
 
 meta def poly.to_pexpr :exmap → poly → tactic pexpr
-| _ (poly.const z) := return $  ``(%%z)
+| _ (poly.const z) := return z.to_pexpr
```

Comment:
This was the root of an error when using `polyrith` on examples over any type other than `rat`. ``` ``(%%z) ``` produces a `pexpr` with type information, so when you elaborate it again, it will be of type `rat`. `rat.to_pexpr` (added above) strips the type information, so it's like a "bare numeral" that will elaborate into whatever type it should have.